### PR TITLE
[LOOP-3325] CGM Signal Loss

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-3325

cgm signal is considered loss if the last communication did not happen within the the stale glucose data limit

### CODE REVIEW

File: LoopUI/ViewModel/CGMStatusHUDViewModel.swift [hunk 1]
```diff
@@ -157,9 +168,44 @@ public class CGMStatusHUDViewModel {
...
     }
+
+    func displayStaleGlucoseValue() {
+        self.glucoseValueString = CGMStatusHUDViewModel.staleGlucoseRepresentation
+        self.glucoseValueTintColor = .label
+        self.trend = nil
+        self.glucoseTrendTintColor = .glucoseTintColor
+        self.manualGlucoseTrendIconOverride = nil
+        if let signalLossHighlight = signalLossHighlight {
+            statusHighlight = signalLossHighlight
```
Comment (darinkrauss) [inline LoopUI/ViewModel/CGMStatusHUDViewModel.swift:175]: [nit] For consistency, consider `self.` prefix here, too.

File: LoopUI/ViewModel/CGMStatusHUDViewModel.swift [hunk 2]
```diff
@@ -157,9 +168,44 @@ public class CGMStatusHUDViewModel {
...
+    }
     
     func setManualGlucoseTrendIconOverride() {
         manualGlucoseTrendIconOverride = storedStatusHighlight?.image ?? UIImage(systemName: "questionmark.circle")
         glucoseTrendTintColor = storedStatusHighlight?.state.color ?? .glucoseTintColor
     }
 }
+
+fileprivate extension CGMStatusHUDViewModel {
+    struct Highlight: DeviceStatusHighlight {
```
Comment (darinkrauss) [inline LoopUI/ViewModel/CGMStatusHUDViewModel.swift:186]: [nit] Consider calling `SignalLossHighlight`.

File: LoopUI/ViewModel/CGMStatusHUDViewModel.swift [hunk 3]
```diff
@@ -101,6 +97,20 @@ public class CGMStatusHUDViewModel {
...
+        didSet {
+            if let signalLossHighlight = signalLossHighlight {
+                statusHighlight = signalLossHighlight
+            }
+        }
+    }
+
+    private var staleGlucoseAge: TimeInterval?
+
+    func setLastCommunicationDate(_ lastCommunicationDate: Date?) {
```
Comment (darinkrauss) [inline LoopUI/ViewModel/CGMStatusHUDViewModel.swift]: [nit] Why not just allow access to `lastCommunicationDate` directly? There's nothing really being done in `setLastCommunicationDate`.

**Review by darinkrauss — APPROVED** [review]
A couple minor nits, but LGTM! 🌞

File: LoopUI/ViewModel/CGMStatusHUDViewModel.swift (hunk 1 above)
Comment (nhamming) [reply LoopUI/ViewModel/CGMStatusHUDViewModel.swift:175]: I'll remove them

**Pull request merged**